go/vt/vtctl/grpcvtctldserver/topo.go

```diff
@@ -32,14 +32,43 @@ import (
 	"vitess.io/vitess/go/vt/proto/vtrpc"
 )
 
-func deleteShard(ctx context.Context, ts *topo.Server, keyspace string, shard string, recursive bool, evenIfServing bool) error {
+func deleteShard(ctx context.Context, ts *topo.Server, keyspace string, shard string, recursive bool, evenIfServing bool, force bool) (err error) {
 	span, ctx := trace.NewSpan(ctx, "VtctldServer.deleteShard")
 	defer span.Finish()
 
 	span.Annotate("keyspace", keyspace)
 	span.Annotate("shard", shard)
 	span.Annotate("recursive", recursive)
 	span.Annotate("even_if_serving", evenIfServing)
+	span.Annotate("force", force)
+
+	lctx, unlock, lerr := ts.LockShard(ctx, keyspace, shard, "DeleteShard")
+	switch {
+	case lerr == nil:
+		// We locked the shard, all good
+		ctx = lctx
+	case !force:
+		return fmt.Errorf("failed to lock %s/%s: %w", keyspace, shard, lerr)
```

Comment:
we should probably add more text to the returned error. Something along the lines of
```
If you really want to delete this shard, re-run with --force
```